---
bool is_fault
string fault 
string recovery 